src/coreclr/jit/importer.cpp

```diff
@@ -12645,6 +12684,11 @@ bool Compiler::impIsAddressInLocal(const GenTree* tree, GenTree** lclVarTreeOut)
         return true;
     }
 
+    if (op->OperIsScalarLocal() && (op->AsLclVarCommon()->GetLclNum() == impInlineRoot()->info.compRetBuffArg))
+    {
+        return true;
+    }
```

Comment:
I don't think it's ok for this to return true without assigning `lclVarTreeOut`.
I think it would be better to add a new function that checks for the property we want, e.g.  `PointsOutsideHeap` or similar. `GenTreeIndir::IsAddressNotOnHeap` could probably be switched to use it as well.